### PR TITLE
feat: testing utils for jest

### PR DESCRIPTION
**Pull request by marklawlor**
# Motivation

Fix: #416
Fix: #161
Fix: #127 

# Execution

Adds a new sub-export `expo-router/testing-library`. This module re-exports `@testing-library/react-native` (required as a optional-dependency) and setups common mocks and test matchers.

It also exports a new function `renderRouter`. This is a wrapper around `render` which can either load your existing Expo Router app and can optionally override routes and layouts. 

See [the docs for usage](https://github.com/expo/router/blob/69c5df288a2c3a10e2e2121076f1aa2a432fb094/docs/docs/lab/jest.md)

### CODE REVIEW

File: packages/expo-router/src/ExpoRoot.tsx [hunk 1]
```diff
@@ -35,21 +36,27 @@ const INITIAL_METRICS = {
...
-      <SafeAreaProvider
-        // SSR support
-        initialMetrics={INITIAL_METRICS}
-      >
-        <ContextNavigator context={context} />
-        {/* Users can override this by adding another StatusBar element anywhere higher in the component tree. */}
-        <StatusBar style="auto" />
-      </SafeAreaProvider>
-    </GestureHandlerRootView>
+    <Head.Provider>
```
Comment (EvanBacon) [inline packages/expo-router/src/ExpoRoot.tsx:39]: This needs to be injected optionally in the entry to ensure it doesn't conflict with the one inside the server renderer.

File: docs/docs/lab/jest.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+};
+```
+
+:::
+
+## Writing a test
+
+```tsx
+// my-component.test.ts
+import { screen } from "expo-router/testing-library";
```
Comment (EvanBacon) [inline docs/docs/lab/jest.md:31]:
```suggestion
import { screen, renderRouter } from "expo-router/testing-library";
```

File: docs/docs/lab/jest.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+type Route =
+  | () => ReactElement
+  | { default: () => ReactELement }
+```
+
+When mocking a router, you can either define it as a function, or an object with a default property
+
+## Matchers
+
+### `.toHavePathname()
```
Comment (EvanBacon) [inline docs/docs/lab/jest.md:107]:
```suggestion
### `.toHavePathname()`
```
Missing backtick.

File: docs/docs/lab/jest.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
+---
+title: Testing with Jest
```
Comment (EvanBacon) [inline docs/docs/lab/jest.md:2]:
```suggestion
title: Testing with Jest
# TODO
sidebar_class_name: hidden
```

**Review by EvanBacon — APPROVED** [review]

**Comment by marklawlor** [comment]
No time frames - more or less when its ready :man_shrugging: I'm still iterating on the design and using it internally to ensure its a useful feature.

**Comment by danielt-ent** [comment]
Ok. Thanks!

**Pull request merged**